packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -105,6 +105,8 @@ export class SingleThreadedWallet
   ): Promise<SingleThreadedWallet> {
     const wallet = new SingleThreadedWallet(walletConfig);
 
+    await wallet.chainService.checkChainId(walletConfig.networkConfiguration.chainNetworkID);
```

Comment:
Have we considered performing this check in the `ChainService` constructor? It might be nice for a consumer of the `ChainService` to not concern itself with the internals of the `ChainService`

Reply:
The problem is that the check needs to be `async`, so we can't do it in the constructor.

Reply:
I suppose we could go down the route of `await ChainService.create()` , i.e. having an async method that calls into the constructor (as we do with the wallet).